platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,442 @@
+   stored here.
+   When a new candidate roster is received, it will be stored in this map, with the `RosterState`'s current candidate
+   roster hash updated.
+
+### On Disk
+
+There will continue to be components used in creating the Roster - such as the private EC (elliptic curve) key, the RSA
+private key and the signing X509Certificate certificate - that will be stored on disk.
+However, it is at the discretion of the Services team to manage the lifecycle of these files, and not the platform.
+
+There will not be any other separate artifacts from this proposal stored directly on disk.
+
+## Services changes (Inversion of control)
+
+The current startup procedure will be altered as follows
+
+- The `config.txt` file will no longer be used by the platform for storing the address book in existing networks. It
+  will no longer be used by the platform code
+- Designating `config.txt` introduces inversion of control. It will be at the exclusive prerogative of Services (the
+  app) going forward. Platform will no longer be responsible for it, and all the platform code that builds an
+  AddressBook from `config.txt` will be removed or refactored and moved into the Services codebase.
+
+### DevOps Workflow changes
+
+The urrent network Transplant procedure is manual.
+DevOps get given a State and `config.txt` file on disk. This is then followed by a software-only upgrade to adopt
+the `config.txt` file using that state. This will change as follows:
+When adding new nodes to an existing network, DevOps will be given a combination of optional State data and Roster with
+the new node in it. Devops will no longer need `config.txt` on new nodes to existing networks.
+One of the State or Roster or both must exist for the network to start.
+
+#### New Transplant Procedure
+
+The app will decide a network transplant sequence based on the following heuristics:
+
+* Network Transplant process == Active Roster AND State provided
+  When both a roster and a state are provided, it signifies a network transplant.
+  The node will discard any existing candidate roster, rotate the active roster to the previous roster,
+  and adopt the provided roster as the new active roster. This way, the provided network state and roster are adopted.
+
+* Genesis Network process == Active Roster, No State provided
+  When a roster is provided, but no pre-existing state exists, it will indicate the creation of a new network.
+  The node will initiate the TSS key generation process (out of scope for this proposal),
+  create a genesis state with the provided roster, and start participating in consensus from round 0.
+* Keep Network Settings (Normal restart) == No Active Roster AND State provided
+  If only a state is provided without a roster, the node will retain its existing network settings, including the active
+  roster.
+  This will be the typical behavior for a node restarting within an established network.
+  ![](Proposed%20Network%20Transplant%20flow.drawio.svg)
```

Comment:
Answering your second question first, the State is not always required. For example, the case of genesis network requires an active roster but no state.

For the first, I'm open to calling it override roster if it helps. @edward-swirldslabs, thoughts?

Reply:
Thanks for the feedback. We have renamed this to 'Genesis Roster' with an accompanying description.

Reply:
Regarding the first question, my point is exactly to make the state a required input argument. Due to the IOC, we can delegate the creation of the genesis state to the app completely. What do you guys think about this?

Reply:
It depends on whether or not there are genesis fields we need to initialize in the state for the platform's purposes.

Reply:
A method will need to be added to PlatformBuilder for genesis as follows:
`void withGenesisRoster(@NonNull final Roster genesisRoster)`
which will be responsible for setting the GenesisRoster when needed. The app can conditionally call this method with a Roster if it determines the startup sequence requires it, though the mechanics of that is upon Services to implement.

Reply:
To reiterate, I specifically question the need for this extra method/extra parameter. I believe we can nuke the genesis roster thing. I believe the app can install a genesis roster directly into the state before passing it to the platform builder on startup. This would simplify our interface with the app.

Would anything break if we do so?

Reply:
Per comments in @edward-swirldslabs 's TSS-Ledger-Id design proposal, it seems that we want to encapsulate roster handling on the Platform side, and from this perspective, accepting a genesis roster as an argument kind of makes sense. So I guess this conversation can be resolved for now.